libs/asset_type.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+"""
+Asset types class
+
+Copyright 2020-2021 Leboncoin
+Licensed under the Apache License, Version 2.0
+Written by Nicolas BEGUIER (nicolas.beguier@adevinta.com)
+"""
+
+from dataclasses import dataclass
+
+from .patterns import Patterns
+
+# Debug
+# from pdb import set_trace as st
+
+@dataclass
+class Location:
+    """
+    Define the asset location
+    """
+    region: str = None
+    vpc: str = None
+    subnet: str = None
+
+class AssetType:
+    """
+    Asset Type
+    """
+    def __init__(self, name: str, public: bool=False):
+        self.name = name
+        self.public = public
+        self.location = Location()
+        self.security_issues = None
+
+    def audit(self, security_config):
+        """
+        This functions is returning an asset_report with security findings,
+        it handles the brief mode output
+        """
+        try:
+            patterns = Patterns(
+                security_config['findings_rules_path'],
+                security_config['severity_levels'],
+                security_config['min_severity'],
+                security_config['max_severity']
+            )
+        except Exception:
+            return
+
+        self.security_issues = patterns.extract_findings(self)
+
+    def display_audit(self, report):
+        """
+        Returns an output of the audit
+        """
+        for issue in self.security_issues:
+            if 'Audit' not in report:
+                report['Audit'] = list()
+            report['Audit'].append(f'[{issue["severity"]}] {issue["title"]}')
+
+    def display_brief_audit(self):
+        """
+        Returns a brief output of the audit
+        """
+        security_issues = dict()
+        output = ''
+        if self.security_issues:
```

Comment:
If `self.security_issues` is set to `[]` as its default value, we don't need to check it here

Reply:
Removed ;)